test/tests/functional/pbs_mom_dynamic_resource_ext.py

```diff
@@ -0,0 +1,212 @@
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 1}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (1kb != (null))"
+        self.scheduler.log_match(c)
+
+    def test_t2(self):
+        """
+        Test for host level consumable float resource
+        with string mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "float", "nh", "/bin/echo abc")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 2}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (2 != (null))"
+        self.scheduler.log_match(c)
+
+    def test_t3(self):
+        """
+        Test for host level string resource
+        with integer mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "string", "h", "/bin/echo 3")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 'abc'}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (abc != 3)"
+        self.scheduler.log_match(c)
+
+    def test_t4(self):
+        """
+        Test for host level string_array resource
+        with float mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "string_array",
+                                 "h", "/bin/echo 3.14")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 'red'}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (red != 3.14)"
+        self.scheduler.log_match(c)
+
+    def test_t5(self):
+        """
+        Tets for host level string_array resource
```

Comment:
any particular reason you are using a string_array resource?  You're really treating it as a string.